### DISCUSSION TITLE
datetime sort with sparse values

### DISCUSSION THREAD

**Discussion by robynm** [TanStack/table#3693] [Q&A] [answered]
How can I use the `datetime` sort type on a column where not every row has a value? I've tried using undefined, null and empty string as the default values, but each time I get the error ```r.getTime is not a function``` when trying to sort on the column.

Thanks!!

**Comment by ZiedHf** [comment, marked as answer]
Try to implement your own sorting method.
This is an example : https://codesandbox.io/s/react-table-sorting-r8q90
See App.js::109

And of course null or undefined do not have getTime so that's an expected error.

**Reply by robynm** [reply]
Thanks for the response. Yes, of course null and undefined don't have a getTime method. I would have thought the sorting function should be able to handle empty values as well, though. I've had very inconsistent results with my own sorting function though, so maybe not.

I've had better results with converting the dates to strings and sorting them that way.
